Draft-Accepted/RFCNNNN-Install-PowerShell.md

```diff
@@ -0,0 +1,174 @@
+avoided by using a native command and closing any open sessions of PowerShell.
+
+## Shipping in Windows
+
+There are several issues that prevent shipping PowerShell 7 in Windows:
+
+- Size
+- Support lifecycle differences
+
+### Size constraints
+
+ A complete installation of PowerShell 7 currently is around 200 MB (unpacked). This is due to
+ including everything (modules and .NET assemblies) to provide parity with what Windows PowerShell
+ 5.1 with .NET Framework provides. There is currently no plan by Windows or the .NET team to ship
+ newer .NET in Windows, so PS7 will need to continue to ship .NET with PS7.
+
+### Support lifecycle
+
+ Windows has a 5+5 year support lifecycle. .NET supports up to 3 years for Long Term Support (LTS)
+ versions. So even if size was not an issue, we cannot check in a version of PS7 with a .NET that is
+ not supported while Windows itself is supported.
+
+### Solution
+
+To solve these issues, the proposal is to ship a new command in Windows that would:
+
+- Be locally discoverable
+- Perform a simple download and install of the latest LTS release of PowerShell
+- Include parameter options to select Stable, Preview, or LTS releases
+
+## Goals/Non-Goals
+
+Goals:
+
+- By default, download and install the latest LTS release to the users CurrentUser Scope
+  - CurrentUser - default installation scope location $env:LOCALAPPDATA\Microsoft\pwsh. If the path
+    exists, remove and replace.
+  - AllUsers - Administrative users may explicit select this scope. AllUsers requires administrative
+    elevation and performs a silent install using the MSI defaults.
+- Allow advanced users to specify channel Stable/Preview/LTS
+
+Non-goals:
+
+- Removal of PowerShell (Remove-PowerShell7)
+- Updating of PowerShell (Update-PowerShell7)
+- User selected versions prior to current (No version parameter)
+- Disconnected scenarios - Users that are air-gapped may download the appropriate release from GitHub.
+- Handling of additional MSI switch options - i.e. Enable PSRemoting
+- Cross platform support
+- Deployments at scale
+
+## Specification
+
+Command name: install-powershell7
+
+The default behavior is to install the zip package silently to CurrentUser scope, with the latest LTS
+release of PowerShell. If the user has administrative
+privilege, they can choose to install the msi package silently to the scope of AllUsers.
+
+- The stable package install location - $env:LOCALAPPDATA\Microsoft\PowerShell-Stable
+  - If path exists, detect installed version and compare:
+    - If versions equal: Provide an error "PowerShell 7 is already installed" and exit.
+    - If versions not equal: Overwrite original installation.
+- The preview package install location - $env:LOCALAPPDATA\Microsoft\PowerShell-Preview
+  - If path exists, detect installed version and compare:
+    - If versions equal: Provide an error "PowerShell 7 is already installed" and exit.
+    - If versions not equal: Overwrite original installation.
```

Comment:
@daxian-dbw / @SteveL-MSFT - Originally we talked about Microsoft\pwsh -- If Microsoft\PowerShell is not a good place, how about Stable: Microsoft\PowerShell-Stable and preview Microsoft\PowerShell-Preview -- Would LTS overwrite stable (and visa versa)?

Reply:
I think AppData\Local\Programs\PowerShell and (PowerShell-Preview) would be the right place.  I see other apps putting themselves there already.  The only downside is that $env:PATH needs to be updated, but we were doing that anyways.

Reply:
Agreed. This looks to be the right location to me.

Reply:
Thank you!